pkg/resources/etcd/statefulset.go

```diff
@@ -314,3 +327,24 @@ func getLauncherImage(data etcdStatefulSetCreatorData) (string, error) {
 	}
 	return data.ImageRegistry(resources.RegistryQuay) + "/kubermatic/etcd-launcher-" + baseTag + ":" + resources.KUBERMATICCOMMIT, nil
 }
+
+func computeReplicas(data etcdStatefulSetCreatorData, set *appsv1.StatefulSet) int {
+	etcdClusterSize := data.Cluster().Spec.EtcdClusterSize
+	if set.Spec.Replicas == nil { // new replicaset
+		return etcdClusterSize
+	}
+	replicas := int(*set.Spec.Replicas)
+	isEtcdHealthy := data.Cluster().Status.ExtendedHealth.Etcd == kubermaticv1.HealthStatusUp
```

Comment:
Can you please move this line to 341, as this line is evaluated and not used if the `etcdClusterSize` == `replicas `